[GTK] Make SystemSettingsProxy a web process supplement and rename it as SystemSettingsManager

carlosgcampos opened the pull request:
#### 8b7209ba250c22f2dfb21d681ca8d706ea3a6c43
<pre>
[GTK] Make SystemSettingsProxy a web process supplement and rename it as SystemSettingsManager
<a href="https://bugs.webkit.org/show_bug.cgi?id=281567">https://bugs.webkit.org/show_bug.cgi?id=281567</a>

Reviewed by Adrian Perez de Castro.

SystemSettingsProxy doesn&apos;t need to be a singleton, it can be a web
process supplement. The proxy is actually in the UI process, sending the
settings to all web processes, so this patch renames SystemSettingsProxy
as SystemSettingsManager and SystemSettingsManager as
SystemSettingsManagerProxy. The web process supplement now connects the
observer in the constructor and initializes the settings from web
process creation parameters later, fixing the initial settings not being
applied before.

* Source/WebCore/platform/glib/SystemSettings.cpp:
(WebCore::SystemSettings::updateSettings):
* Source/WebCore/platform/glib/SystemSettings.h:
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/glib/SystemSettingsManagerProxy.cpp: Renamed from Source/WebKit/UIProcess/glib/SystemSettingsManager.cpp.
(WebKit::SystemSettingsManagerProxy::themeName const):
(WebKit::SystemSettingsManagerProxy::darkMode const):
(WebKit::SystemSettingsManagerProxy::fontName const):
(WebKit::SystemSettingsManagerProxy::xftAntialias const):
(WebKit::SystemSettingsManagerProxy::xftHinting const):
(WebKit::SystemSettingsManagerProxy::xftHintStyle const):
(WebKit::SystemSettingsManagerProxy::xftRGBA const):
(WebKit::SystemSettingsManagerProxy::xftDPI const):
(WebKit::SystemSettingsManagerProxy::cursorBlink const):
(WebKit::SystemSettingsManagerProxy::cursorBlinkTime const):
(WebKit::SystemSettingsManagerProxy::primaryButtonWarpsSlider const):
(WebKit::SystemSettingsManagerProxy::overlayScrolling const):
(WebKit::SystemSettingsManagerProxy::enableAnimations const):
(WebKit::SystemSettingsManagerProxy::initialize):
(WebKit::SystemSettingsManagerProxy::settingsDidChange):
* Source/WebKit/UIProcess/glib/SystemSettingsManagerProxy.h: Renamed from Source/WebKit/UIProcess/glib/SystemSettingsManager.h.
* Source/WebKit/UIProcess/gtk/Display.cpp:
(WebKit::Display::Display):
* Source/WebKit/UIProcess/gtk/SystemSettingsManagerProxyGtk.cpp: Renamed from Source/WebKit/UIProcess/gtk/SystemSettingsManagerGtk.cpp.
(WebKit::SystemSettingsManagerProxy::themeName const):
(WebKit::SystemSettingsManagerProxy::darkMode const):
(WebKit::SystemSettingsManagerProxy::fontName const):
(WebKit::SystemSettingsManagerProxy::xftAntialias const):
(WebKit::SystemSettingsManagerProxy::xftHinting const):
(WebKit::SystemSettingsManagerProxy::xftHintStyle const):
(WebKit::SystemSettingsManagerProxy::xftRGBA const):
(WebKit::SystemSettingsManagerProxy::xftDPI const):
(WebKit::SystemSettingsManagerProxy::cursorBlink const):
(WebKit::SystemSettingsManagerProxy::cursorBlinkTime const):
(WebKit::SystemSettingsManagerProxy::primaryButtonWarpsSlider const):
(WebKit::SystemSettingsManagerProxy::overlayScrolling const):
(WebKit::SystemSettingsManagerProxy::enableAnimations const):
(WebKit::SystemSettingsManagerProxy::SystemSettingsManagerProxy):
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::initializeProcess):
* Source/WebKit/WebProcess/glib/SystemSettingsManager.cpp: Added.
(WebKit::SystemSettingsManager::SystemSettingsManager):
(WebKit::SystemSettingsManager::~SystemSettingsManager):
(WebKit::SystemSettingsManager::supplementName):
(WebKit::SystemSettingsManager::initialize):
(WebKit::SystemSettingsManager::didChange):
* Source/WebKit/WebProcess/glib/SystemSettingsManager.h: Renamed from Source/WebKit/WebProcess/glib/SystemSettingsProxy.h.
* Source/WebKit/WebProcess/glib/SystemSettingsManager.messages.in: Renamed from Source/WebKit/WebProcess/glib/SystemSettingsProxy.messages.in.
* Source/WebKit/WebProcess/glib/SystemSettingsProxy.cpp: Removed.
* Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
(WebKit::WebProcess::platformInitializeProcess):
(WebKit::WebProcess::platformInitializeWebProcess):
(WebKit::WebProcess::platformTerminate):

Canonical link: <a href="https://commits.webkit.org/285320@main">https://commits.webkit.org/285320@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3b2461785d2a37688a8f86b844253ff500c658ae

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/72037 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/51457 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/24822 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/76200 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/23246 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/74152 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/59258 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/23066 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/56835 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/15336 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/75104 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/46654 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/62057 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/37271 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/43317 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/19530 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/21596 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/65220 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/19891 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/77881 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/16277 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/19063 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/65310 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/16323 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/62080 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/64565 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/12762 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/6406 "Found 1 new test failure: imported/w3c/web-platform-tests/preload/preload-type-match.html (failure)") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11089 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/47255 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/2039 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/48324 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/49611 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/48068 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WebKit/WebProcess/glib/SystemSettingsManager.h:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021 Purism SPC
+ * Copyright (C) 2024 Igalia S.L.
```
Shouldn't we keep the Purism copyright line along with the added Igalia one?

carlosgcampos replied:
Oops, yes.

aperezdc reviewed (approved):
LGTM, thanks for the fix!

carlosgcampos committed:
[GTK] Make SystemSettingsProxy a web process supplement and rename it as SystemSettingsManager
https://bugs.webkit.org/show_bug.cgi?id=281567

Reviewed by Adrian Perez de Castro.

SystemSettingsProxy doesn't need to be a singleton, it can be a web
process supplement. The proxy is actually in the UI process, sending the
settings to all web processes, so this patch renames SystemSettingsProxy
as SystemSettingsManager and SystemSettingsManager as
SystemSettingsManagerProxy. The web process supplement now connects the
observer in the constructor and initializes the settings from web
process creation parameters later, fixing the initial settings not being
applied before.

* Source/WebCore/platform/glib/SystemSettings.cpp:
(WebCore::SystemSettings::updateSettings):
* Source/WebCore/platform/glib/SystemSettings.h:
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/glib/SystemSettingsManagerProxy.cpp: Renamed from Source/WebKit/UIProcess/glib/SystemSettingsManager.cpp.
(WebKit::SystemSettingsManagerProxy::themeName const):
(WebKit::SystemSettingsManagerProxy::darkMode const):
(WebKit::SystemSettingsManagerProxy::fontName const):
(WebKit::SystemSettingsManagerProxy::xftAntialias const):
(WebKit::SystemSettingsManagerProxy::xftHinting const):
(WebKit::SystemSettingsManagerProxy::xftHintStyle const):
(WebKit::SystemSettingsManagerProxy::xftRGBA const):
(WebKit::SystemSettingsManagerProxy::xftDPI const):
(WebKit::SystemSettingsManagerProxy::cursorBlink const):
(WebKit::SystemSettingsManagerProxy::cursorBlinkTime const):
(WebKit::SystemSettingsManagerProxy::primaryButtonWarpsSlider const):
(WebKit::SystemSettingsManagerProxy::overlayScrolling const):
(WebKit::SystemSettingsManagerProxy::enableAnimations const):
(WebKit::SystemSettingsManagerProxy::initialize):
(WebKit::SystemSettingsManagerProxy::settingsDidChange):
* Source/WebKit/UIProcess/glib/SystemSettingsManagerProxy.h: Renamed from Source/WebKit/UIProcess/glib/SystemSettingsManager.h.
* Source/WebKit/UIProcess/gtk/Display.cpp:
(WebKit::Display::Display):
* Source/WebKit/UIProcess/gtk/SystemSettingsManagerProxyGtk.cpp: Renamed from Source/WebKit/UIProcess/gtk/SystemSettingsManagerGtk.cpp.
(WebKit::SystemSettingsManagerProxy::themeName const):
(WebKit::SystemSettingsManagerProxy::darkMode const):
(WebKit::SystemSettingsManagerProxy::fontName const):
(WebKit::SystemSettingsManagerProxy::xftAntialias const):
(WebKit::SystemSettingsManagerProxy::xftHinting const):
(WebKit::SystemSettingsManagerProxy::xftHintStyle const):
(WebKit::SystemSettingsManagerProxy::xftRGBA const):
(WebKit::SystemSettingsManagerProxy::xftDPI const):
(WebKit::SystemSettingsManagerProxy::cursorBlink const):
(WebKit::SystemSettingsManagerProxy::cursorBlinkTime const):
(WebKit::SystemSettingsManagerProxy::primaryButtonWarpsSlider const):
(WebKit::SystemSettingsManagerProxy::overlayScrolling const):
(WebKit::SystemSettingsManagerProxy::enableAnimations const):
(WebKit::SystemSettingsManagerProxy::SystemSettingsManagerProxy):
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::initializeProcess):
* Source/WebKit/WebProcess/glib/SystemSettingsManager.cpp: Added.
(WebKit::SystemSettingsManager::SystemSettingsManager):
(WebKit::SystemSettingsManager::~SystemSettingsManager):
(WebKit::SystemSettingsManager::supplementName):
(WebKit::SystemSettingsManager::initialize):
(WebKit::SystemSettingsManager::didChange):
* Source/WebKit/WebProcess/glib/SystemSettingsManager.h: Renamed from Source/WebKit/WebProcess/glib/SystemSettingsProxy.h.
* Source/WebKit/WebProcess/glib/SystemSettingsManager.messages.in: Renamed from Source/WebKit/WebProcess/glib/SystemSettingsProxy.messages.in.
* Source/WebKit/WebProcess/glib/SystemSettingsProxy.cpp: Removed.
* Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
(WebKit::WebProcess::platformInitializeProcess):
(WebKit::WebProcess::platformInitializeWebProcess):
(WebKit::WebProcess::platformTerminate):

Canonical link: https://commits.webkit.org/285320@main
